pennylane/debugging/operation_checker.py

```diff
@@ -0,0 +1,833 @@
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``get_matrix``. ``None`` if no matrix is defined via ``get_matrix``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    return wrap_op_method(instance, "get_matrix", MatrixUndefinedError)()
+
+
+def matrix_from_sparse_matrix(op, par, wires):
+    r"""Get the matrix of an operation, using ``sparse_matrix``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``sparse_matrix``. ``None`` if no matrix is defined via ``sparse_matrix``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    smat = wrap_op_method(instance, "sparse_matrix", SparseMatrixUndefinedError)()
+    if smat is None or isinstance(smat, Exception):
+        return smat
+    return qml.matrix(smat)
+
+
+def matrix_from_terms(op, par, wires):
+    r"""Get the matrix of an operation, using its ``terms``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``terms``. ``None`` if no terms are defined via ``terms``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    terms = wrap_op_method(instance, "terms", TermsUndefinedError)()
+    if terms is None or isinstance(terms, Exception):
+        return terms
+
+    return np.sum([c * qml.matrix(op) for c, op in zip(*terms)], axis=0)
+
+
+def matrix_from_decomposition(op, par, wires):
+    r"""Get the matrix of an operation, using its ``decomposition``.
+
+    Args:
+        op (type): Operation type to obtain the matrix for
+        par (array_like): Parameters of the operation
+        wires (.wires.Wires): Wires of the operation
+
+    Returns:
+        object: Matrix of the operation if it is defined and no problem occured
+        with ``expand``. ``None`` if no decomposition is defined via ``expand``
+        or ``Exception`` if an error occured.
+    """
+    instance = op(*par, wires=wires)
+    tape = wrap_op_method(instance, "expand", DecompositionUndefinedError)()
+    return qml.matrix(tape, wire_order=wires) if isinstance(tape, qml.tape.QuantumTape) else tape
+
+
+def matrix_from_single_qubit_rot_angles(op, par, wires):
+    r"""Get the matrix of an operation, using its ``single_qubit_rot_angles``.
```

Comment:
Might be worth tagging @mariaschuld here, as I don't know if `single_qubit_rot_angles` are going to survive!

Reply:
Yes, that is an anachronism that no one had the time to turn into an "attribute" (since it is a bit different from the other attributes, it contains numeric info...)